### ISSUE TITLE
Integrate IntervalSets.jl

### ISSUE THREAD

**Issue by jmejia8** [jmejia8/SearchSpaces.jl#4]
Integreate IntervalSets.jl due to https://github.com/jmejia8/Metaheuristics.jl/issues/82

**Comment by aplavin** [comment]
Indeed, intervals are so fundamental, that lots of packages converged on using a single implementation (`IntervalSets.jl`) to gain this interop and familiarity.

Potentially, you can also reuse other domains from `DomainSets.jl` here, but they are less of an actual issue.
